index.js

```diff
@@ -71,12 +43,30 @@ LocaleProcessor.prototype.transform = function (localeName, fields, pluralFn) {
 LocaleProcessor.prototype.processString = function (inputString, filename) {
     var localeName = path.basename(filename, path.extname(filename));
 
-    var cldr = process(localeName, {
-        plurals: true,
-        fields:  ['year', 'month', 'day', 'hour', 'minute', 'second']
+    localeName = cldrLocales.locales.normalize(localeName);
+
+    var out   = {};
+    var parts = localeName.split('-');
+    
+    var data = cldr({
+        locales:        [localeName],
+        pluralRules:    true,
+        relativeFields: ['year', 'month', 'day', 'hour', 'minute', 'second']
+    });
+
+    parts.map(function(part, idx) {
+        /* 
+         * if the locale is 'zh-Hant-TW' this map will build an
+         * array of strings ['zh', 'zh-Hant', 'zh-Hant-TW']
+         * which is used to walk the CLDR results to progressively
+         * build up the results object
+         */
```

Comment:
Update: I am going to have to make much larger changes to accommodate this new result object since I am no longer building one ES6 module per locale, I need to create an ES6 module per each part of the result set to prevent duplication.